src/utils/BitcoinTx.sol

```diff
@@ -211,18 +217,40 @@ library BitcoinTx {
         uint256 outputStartingIndex;
         // The number of outputs in the transaction.
         uint256 outputsCount;
+        // The type of data after the OP_RETURN code.
+        OutputType outputType;
     }
 
     /// @notice Represents an outcome of the Bitcoin transaction
-    ///         outputs processing.
+    ///         outputs processing potentially containing evm address.
     struct TxOutputsInfo {
         // Sum of all outputs values spending to the output.
         uint64 value;
         // Optional EVM address specified in OP_RETURN.
         address evmAddress;
     }
 
-    /// @notice Processes the Bitcoin transaction output vector.
+    /// @notice Represents an outcome of the Bitcoin transaction
+    ///         outputs processing  potentially containing 32 byte hash.
+    struct TxOutputsInfoHash {
+        // Sum of all outputs values spending to the output.
+        uint64 value;
+        // Optional hash specified in OP_RETURN.
+        bytes32 hash;
```

Comment:
I think add this to `TxOutputsInfo` or rename that to `TxOutputsInfoAddress` but since both are allowed to be zero I think we can just use a single struct.